components/scrubber/scrubber.go

```diff
@@ -94,11 +106,26 @@ func (s *scrubberImpl) JSON(msg json.RawMessage) (json.RawMessage, error) {
 
 // KeyValue implements Scrubber
 func (s *scrubberImpl) KeyValue(key string, value string) (sanitisedValue string) {
-	f, ok := s.FieldIndex[strings.ToLower(key)]
-	if !ok {
+	sanitisatiser := s.GetSanitisatiser(key)
+	if sanitisatiser == nil {
 		return value
 	}
-	return f(value)
+	return sanitisatiser(value)
+}
+
+// GetSanitisatiser implements
+func (s *scrubberImpl) GetSanitisatiser(key string) Sanitisatiser {
+	if f, ok := s.FieldIndex[key]; ok {
+		return f
+	}
+	for regexp, f := range s.RegexpIndex {
```

Comment:
nit: `regexp` is the name of the package as well which makes this a bit clunky. `expr` could be an alternative